src/v/security/acl.h

```diff
@@ -441,6 +521,57 @@ class resource_pattern_filter {
           std::nullopt, std::nullopt, std::nullopt);
         return filter;
     }
+    friend void read_nested(
+      iobuf_parser& in,
+      resource_pattern_filter& filter,
+      size_t const bytes_left_limit) {
+        using serde::read_nested;
+
+        read_nested(in, filter._resource, bytes_left_limit);
+        read_nested<std::optional<ss::sstring>>(
+          in, filter._name, bytes_left_limit);
+
+        auto pattern = read_nested<std::optional<serialized_pattern_type>>(
+          in, bytes_left_limit);
+
+        if (!pattern) {
+            filter._pattern = std::nullopt;
+            return;
+        }
+
+        switch (*pattern) {
+        case serialized_pattern_type::literal:
+            filter._pattern = security::pattern_type::literal;
+            break;
+
+        case serialized_pattern_type::prefixed:
+            filter._pattern = security::pattern_type::prefixed;
+            break;
+        case serialized_pattern_type::match:
+            filter._pattern
+              = security::resource_pattern_filter::pattern_match{};
```

Comment:
I suspect this shouldn't be default constructed.

Reply:
this is just a placeholder it is exactly the same serialization code as in ADL.